### PR TITLE
[RFR] Cloud Intel / Reports widgetastic conversion

### PR DESCRIPTION
**Pull request by quarckster**
PRT failed tests:
test_custom_report_crud[basic01.yaml] - PASSED locally
test_schedule_crud[basic01.yaml] - failed due https://bugzilla.redhat.com/show_bug.cgi?id=1441101
test_shuffle_top_level[EvmGroup-super_administrator] - PASSED locally
test_shuffle_first_level[EvmGroup-super_administrator] - PASSED locally
test_widgets_on_dashboard - requires https://github.com/ManageIQ/integration_tests/blob/master/cfme/dashboard.py to be converted in widgetastic
~~Requires https://github.com/RedHatQE/widgetastic.patternfly/pull/1 to be merged~~
Requires #5039 to be merged

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 6c123d43d0a4846bd7b8cd949086b748a9cc559f:

`cfme/intelligence/reports/dashboards.py`:
- :bangbang: Line 26:16: **E999** *SyntaxError: invalid syntax*
- :bangbang: Line 128:1: **E122** *continuation line missing indentation or outdented*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c78cd18209794f77dc8db5ed8af32bd4dd83e42a:

`cfme/intelligence/reports/dashboards.py`:
- :bangbang: Line 26:16: **E999** *SyntaxError: invalid syntax*
- :bangbang: Line 128:1: **E122** *continuation line missing indentation or outdented*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 192db295f8113b43e2f77c092b25006a2697f42e:

`cfme/intelligence/reports/widgets/__init__.py`:
- :red_circle: Line 26:9: **F821** *undefined name 'wait_for'*
- :red_circle: Line 29:23: **F821** *undefined name 'toolbar'*

`widgetastic_manageiq.py`:
- :heavy_exclamation_mark: Line 1136:57: **W291** *trailing whitespace*
- :red_circle: Line 1210:65: **F821** *undefined name 'values_to_add'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3c856d889c7350d5f14c86d1cb32d14c91a8efaf:

`cfme/intelligence/reports/widgets/__init__.py`:
- :red_circle: Line 26:9: **F821** *undefined name 'wait_for'*
- :red_circle: Line 29:23: **F821** *undefined name 'toolbar'*

`widgetastic_manageiq.py`:
- :heavy_exclamation_mark: Line 1136:57: **W291** *trailing whitespace*
- :red_circle: Line 1210:65: **F821** *undefined name 'values_to_add'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 7e534c95f3b93f3549e643b60edb022b9158c172:

`cfme/intelligence/reports/widgets/__init__.py`:
- :red_circle: Line 26:9: **F821** *undefined name 'wait_for'*
- :red_circle: Line 29:23: **F821** *undefined name 'toolbar'*

`widgetastic_manageiq.py`:
- :heavy_exclamation_mark: Line 1136:57: **W291** *trailing whitespace*
- :red_circle: Line 1210:65: **F821** *undefined name 'values_to_add'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 29f02cca77e060e04bdddf96e9af35ef7977e8a3:

`cfme/intelligence/reports/reports.py`:
- :red_circle: Line 7:1: **F401** *'navmazing.NavigateToAttribute' imported but unused*
- :red_circle: Line 8:1: **F401** *'utils.appliance.implementations.ui.CFMENavigateStep' imported but unused*
- :red_circle: Line 8:1: **F401** *'utils.appliance.implementations.ui.navigator' imported but unused*
- :red_circle: Line 12:1: **F401** *'widgetastic_manageiq.SummaryFormItem' imported but unused*
- :red_circle: Line 12:1: **F401** *'widgetastic_manageiq.DashboardWidgetsPicker' imported but unused*
- :red_circle: Line 109:33: **F821** *undefined name 'AllReportsView'*
- :red_circle: Line 123:33: **F821** *undefined name 'CustomReportDetailsView'*
- :red_circle: Line 142:37: **F821** *undefined name 'AllCustomReportsView'*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 1a1795bc4ab6a01ce4f7625e050b217b96086430:

`cfme/intelligence/reports/reports.py`:
- :red_circle: Line 13:1: **F401** *'widgetastic_manageiq.SummaryFormItem' imported but unused*
- :red_circle: Line 13:1: **F401** *'widgetastic_manageiq.DashboardWidgetsPicker' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit db3df5717b52a5946ed72a57a9ec21723f0e7430:

`cfme/intelligence/reports/reports.py`:
- :red_circle: Line 114:11: **N801** *class names should use CapWords convention*
- :red_circle: Line 119:11: **N801** *class names should use CapWords convention*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,98 @@
+<<<<<<< HEAD
```
Comment (mfalesni) [inline requirements.txt]: Bad rebase :)

**Comment by dajoRH** [comment]
Lint report for commit 02d2a4b17f280be6ecc9ecd8b4aea68f115b3f52:

`cfme/intelligence/reports/reports.py`:
- :bangbang: Line 431:17: **E131** *continuation line unaligned for hanging indent*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 64045f5d3a652c255357bc4b4411f5cee0970870:

`cfme/intelligence/reports/reports.py`:
- :bangbang: Line 596:8: **E122** *continuation line missing indentation or outdented*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/intelligence/reports/widgets/chart_widgets.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+class NewChartWidgetView(NewDashboardWidgetView, ChartWidgetFormCommon):
+    pass
+
+
+class EditChartWidgetView(EditDashboardWidgetView, ChartWidgetFormCommon):
+    pass
+
+
+class ChartWidget(Widget):
```
Comment (mfalesni) [inline cfme/intelligence/reports/widgets/chart_widgets.py]: Can you cahnge the Widget class name to ManageIQWidget or something? This confuses with WT widget :)

File: widgetastic_manageiq.py [hunk 3]
```diff
@@ -1761,3 +1762,109 @@ def fill(self, value):
...
+        else:
+            return []
+
+    def clear(self):
+        for el in self.browser.elements(".//a[@title='Remove this Shortcut']"):
+            self.browser.click(el)
+
+    def fill(self, values):
+        dict_values = None
+        if isinstance(values, basestring):
```
Comment (mfalesni) [inline widgetastic_manageiq.py]: Use six.basestring here for compatibility with both pythons.

File: widgetastic_manageiq.py [hunk 4]
```diff
@@ -1761,3 +1762,109 @@ def fill(self, value):
...
+        self.names_locator = names_locator
+        self.remove_locator = remove_locator
+        self.select_id = select_id
+
+    def add_widget(self, widget):
+        self.select.fill(widget)
+
+    def remove_widget(self, widget):
+        self.browser.click(self.remove_locator.format(quote(widget)))
+        self.browser.plugin.ensure_page_safe()
```
Comment (mfalesni) [inline widgetastic_manageiq.py]: click in browser calls this by default.

File: widgetastic_manageiq.py [hunk 5]
```diff
@@ -1761,3 +1762,109 @@ def fill(self, value):
...
+    def __init__(self, parent, id, names_locator=None, remove_locator=None,
+                 select_id=None, logger=None):
+        View.__init__(self, parent=parent, logger=logger)
+        self.id = id
+        self.names_locator = names_locator
+        self.remove_locator = remove_locator
+        self.select_id = select_id
+
+    def add_widget(self, widget):
+        self.select.fill(widget)
```
Comment (mfalesni) [inline widgetastic_manageiq.py]: I think fill should call add and not reverse, but your decision :)

Comment (quarckster) [inline widgetastic_manageiq.py]: I don't think so, here the parent widget calls `fill()` method of descendant widget.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b428bea361a10e854596cfb75eff97916208e03d:

`cfme/intelligence/reports/menus.py`:
- :bangbang: Line 28:17: **E999** *SyntaxError: invalid syntax*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Pull request closed without merging**